app/client/src/components/shared/WaterbodyInfo.tsx

```
@@ -1078,6 +1116,570 @@ function MapPopup({
+    if (!projectionLoaded) return;
+
+    const cyanImageLayer = mapView.map.findLayerById('cyanImages');
+    if (!cyanImageLayer || !isMediaLayer(cyanImageLayer)) return;
+
+    const currentDate = new Date(selectedDate);
+    const imageUrl = `${services.data.cyan.images}/?OBJECTID=${
+      attributes.oid ?? attributes.OBJECTID
+    }&year=${currentDate.getFullYear()}&day=${getDayOfYear(currentDate)}`;
+    const propertiesUrl = `${services.data.cyan.properties}/?OBJECTID=${
+      attributes.oid ?? attributes.OBJECTID
+    }`;
+
+    const abortController = new AbortController();
+    const timeout = 60_000;
+    const imageTimeout = setTimeout(() => abortController.abort(), timeout);
+
+    cyanImageLayer.source.elements.removeAll();
+    setImageStatus('pending');
+    const imagePromise = fetch(imageUrl, { signal: abortController.signal });
+    const propsPromise = fetchCheck(propertiesUrl, abortController.signal);
+    Promise.all([imagePromise, propsPromise])
+      .then(([imageRes, propsRes]) => {
+        if (imageRes.headers.get('Content-Type') !== 'image/png') {
+          setImageStatus('idle');
+          return;
+        }
+
+        imageRes.blob().then((blob) => {
+          const image = new Image();
+          image.src = URL.createObjectURL(blob);
+          image.onload = () => setImageStatus('success');
+          const imageElement = new ImageElement({
+            image,
+            georeference: new ExtentAndRotationGeoreference({
+              extent: new Extent({
+                spatialReference: SpatialReference.WGS84,
+                xmin: propsRes.properties.x_min,
+                xmax: propsRes.properties.x_max,
+                ymin: propsRes.properties.y_min,
+                ymax: propsRes.properties.y_max,
+              }),
+            }),
+          });
+          cyanImageLayer.source.elements.add(imageElement);
+        });
+      })
+      .catch((err) => {
+        setImageStatus('failure');
```

Can you display an error message on the screen when the `imageStatus === 'failure'`? Currently, it just silently fails if either the `https://qed.epa.gov/waterbody/image` or `https://qed.epa.gov/waterbody/properties` services are down.

---

Good call, I'll add that.